packages/block-editor/src/components/block-switcher/block-transformations-menu.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { MenuGroup, MenuItem } from '@wordpress/components';
+
+const BlockTransformationsMenu = ( {
+	possibleBlockTransformations,
+	onSelect,
+} ) => {
+	return (
+		<MenuGroup>
+			<div className="block-editor-block-switcher__label">
+				{ __( 'Transform to' ) }
+			</div>
+			{ possibleBlockTransformations.map( ( item ) => {
+				const { name, icon: { src: iconSrc } = {}, title } = item;
+				return (
+					<MenuItem
+						key={ name }
+						icon={ iconSrc }
```

Comment:
Did you try wrapping an icon with `BlockIcon` component? 
https://github.com/WordPress/gutenberg/tree/master/packages/block-editor/src/components/block-icon

Reply:
There is similar code in the another file you modified:

```js
icon={ <BlockIcon icon={ icon } showColors /> }
```

Reply:
Yes, but the new parent component used, was expecting a `string` value.

Reply:
Is anything blocking from changing it to accept component?

Reply:
Changed it to `BlockIcon`. Though there was no problem using it... Probably my initial problem was related to some first iterations

Reply:
@gziolo What exactly is the purpose of the `BlockIcon` component?